components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/dao/RequestObjectDAOImplTest.java

```diff
@@ -55,12 +87,135 @@ private void testInsertRequestObject() throws IdentityOAuth2Exception {
+            Assert.assertEquals(resultSize, 0);
+        }
+    }
+
+    @Test void testUpdateRequestObjectReferenceCodeToToken() throws Exception {
+
+        requestObjectDAO.insertRequestObjectData(consumerKey, sessionDataKey,
                 requestedEssentialClaims);
-        requestObjectDAO.getRequestedClaims("d43e8da324a33bdc941b9b95cad6a6a2", true);
-        Assert.assertEquals(requestObjectDAO.getRequestedClaims("code1", true).size(), 0);
+        insertCodeId(codeId, 1);
+        requestObjectDAO.updateRequestObjectReferencebyCodeId(sessionDataKey, codeId);
+        requestObjectDAO.updateRequestObjectReferenceCodeToToken(codeId, tokenId);
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection()) {
+            String query = "SELECT * FROM IDN_OIDC_REQ_OBJECT_REFERENCE WHERE CODE_ID=? AND TOKEN_ID=?";
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, codeId);
+            statement.setString(2, tokenId);
+            ResultSet resultSet = statement.executeQuery();
+            int resultSize = 0;
+            if (resultSet.next()) {
+                resultSize = resultSet.getRow();
+            }
+            Assert.assertEquals(resultSize, 1);
+        }
+    }
+
+    protected void insertCodeId(String codeId, int consumerKeyId) throws Exception {
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection()) {
+            String sql = "INSERT INTO IDN_OAUTH2_AUTHORIZATION_CODE (CODE_ID, CONSUMER_KEY_ID) VALUES (?,?)";
+            PreparedStatement ps = connection.prepareStatement(sql);
+            ps.setString(1, codeId);
+            ps.setInt(2, consumerKeyId);
+            ps.execute();
+            IdentityDatabaseUtil.commitTransaction(connection);
+        } catch (SQLException e) {
+            log.error("Error when inserting codeID object.", e);
+            throw new IdentityOAuth2Exception("Error when inserting codeID", e);
+        }
+    }
+
+    protected String[] getData(String sessionDataKey) throws Exception {
+
+        String[] dataMap = new String[3];
+        ResultSet resultSet = null;
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection()) {
+            String sql = "SELECT CONSUMER_KEY_ID, CODE_ID, TOKEN_ID FROM IDN_OIDC_REQ_OBJECT_REFERENCE WHERE " +
+                    "SESSION_DATA_KEY=?";
+
+            PreparedStatement prepStmt = connection.prepareStatement(sql);
+            prepStmt.setString(1, sessionDataKey);
+            resultSet = prepStmt.executeQuery();
+
+            while (resultSet.next()) {
+                dataMap = new String[]{resultSet.getString(1), resultSet.getString(2), resultSet.getString(3)};
+            }
+        } catch (SQLException e) {
+            log.error("Error when retrieving inserted request object.", e);
+            throw new IdentityOAuth2Exception("Error when retrieving request object", e);
+        }
+        return dataMap;
```

Comment:
Lets refactor this method to be more meaningful. Introduce a model and give 1st class support for the data retrieved from the db instead of putting it to an array.

Also do we need to make this method protected?

Reply:
Addressed with 1c087f83ec0cada23b770b13bc32829d7e4ef83e